opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/core/ExpressionScript.java

```diff
@@ -107,7 +113,15 @@ public Object visitParse(ParseExpression node, Set<ReferenceExpression> context)
 
   private OpenSearchExprValueFactory buildValueFactory(Set<ReferenceExpression> fields) {
     Map<String, OpenSearchDataType> typeEnv = fields.stream().collect(toMap(
-        ReferenceExpression::getAttr, e -> OpenSearchDataType.of(e.type())));
+        ReferenceExpression::getAttr, e -> {
+          if (e.type() == TIMESTAMP
```

Comment:
Same as Andrew said here: https://github.com/Bit-Quill/opensearch-project-sql/pull/268/files#r1207314626

Reply:
Done in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/5f34b391f988e4f2e179a68bce688abc1dc15914